### PR TITLE
Add resequence message store support for devstudio 

### PR DESCRIPTION
**Pull request by Drifftr**
## Purpose
This adds the resequence support for Developer studio

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: plugins/org.wso2.developerstudio.eclipse.artifact.messagestore/src/org/wso2/developerstudio/eclipse/artifact/messagestore/ui/wizard/MessageStoreCreationWizard.java [hunk 1]
```diff
@@ -365,6 +379,56 @@ private String getTemplateContent(){
...
+						parameters.put(STORE_RESEQUENCE_CONNECTION_URL, messageStoreModel.getResequenceURL());
+					}
+					if (StringUtils.isNotBlank(messageStoreModel.getResequenceUser())) {
+						parameters.put(STORE_RESEQUENCE_USERNAME, messageStoreModel.getResequenceUser());
+					}
+					if (StringUtils.isNotBlank(messageStoreModel.getResequencePassword())) {
+						parameters.put(STORE_RESEQUENCE_PASSWORD, messageStoreModel.getResequencePassword());
+					}
+				} else if (JDBCConnectionInformationType.CARBON_DATASOURCE.toString().equals(resequenceConnectionInformation)) {
+					if (StringUtils.isNotBlank(messageStoreModel.getResequenceDatasourceName())) {
```
Comment (milindaperera) [inline plugins/org.wso2.developerstudio.eclipse.artifact.messagestore/src/org/wso2/developerstudio/eclipse/artifact/messagestore/ui/wizard/MessageStoreCreationWizard.java]: we can merge these two else if and if blocks to one

File: plugins/org.wso2.developerstudio.eclipse.artifact.messagestore/src/org/wso2/developerstudio/eclipse/artifact/messagestore/validator/MessageStoreFieldController.java [hunk 2]
```diff
@@ -119,6 +126,33 @@ public void validate(String key, Object value,
...
+				CommonFieldValidator.isValidUrl(value.toString(), "JDBC");
+			}
+		} else if (key.equals(FIELD_RESEQUENCER_USER)) {
+			if (resequence && !isCarbonDataSource) {
+				CommonFieldValidator.validateRequiredField(value, "Resequence User cannot be empty");
+			}
+		} else if (key.equals(FIELD_RESEQUENCER_DATASOURCE_NAME)) {
+			if (resequence && isCarbonDataSource) {
+				CommonFieldValidator.validateRequiredField(value, "Resequence Datasource Name cannot be empty");
+			}
```
Comment (milindaperera) [inline plugins/org.wso2.developerstudio.eclipse.artifact.messagestore/src/org/wso2/developerstudio/eclipse/artifact/messagestore/validator/MessageStoreFieldController.java]: We can merge many and reduce nested if blocks above

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/MessageStoreDeserializer.java [hunk 3]
```diff
@@ -343,6 +366,72 @@
...
+								} else if ("false".equals(value)) {
+									executeSetValueCommand(MESSAGE_STORE__ENABLE_PRODUCER_GUARANTEED_DELIVERY, false);
+								}
+							}
+						} else if (param.getKey().equals(STORE_FAILOVER_MESSAGE_STORE_NAME)) {
+							if (StringUtils.isNotBlank(value)) {
+								executeSetValueCommand(MESSAGE_STORE__FAILOVER_MESSAGE_STORE, value);
+							}
+						} else if (param.getKey().equals(STORE_RESEQUENCE_POLLING_COUNT)) {
+							if (StringUtils.isNotBlank(value)) {
```
Comment (milindaperera) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/MessageStoreDeserializer.java]: nested single if block around here too

File: plugins/org.wso2.developerstudio.eclipse.artifact.messagestore/src/org/wso2/developerstudio/eclipse/artifact/messagestore/ui/wizard/MessageStoreCreationWizard.java (hunk 1 above)
Comment (Drifftr) [reply plugins/org.wso2.developerstudio.eclipse.artifact.messagestore/src/org/wso2/developerstudio/eclipse/artifact/messagestore/ui/wizard/MessageStoreCreationWizard.java]: Fixed with 7c27579

File: plugins/org.wso2.developerstudio.eclipse.artifact.messagestore/src/org/wso2/developerstudio/eclipse/artifact/messagestore/validator/MessageStoreFieldController.java (hunk 2 above)
Comment (Drifftr) [reply plugins/org.wso2.developerstudio.eclipse.artifact.messagestore/src/org/wso2/developerstudio/eclipse/artifact/messagestore/validator/MessageStoreFieldController.java]: Fixed with 7c27579

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/MessageStoreDeserializer.java (hunk 3 above)
Comment (Drifftr) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/MessageStoreDeserializer.java]: Fixed with 7c27579

**Pull request merged**